### PR TITLE
Add Scotto34 (PCB Edition)

### PR DESCRIPTION
**Pull request by joe-scotto**
## Description

Add the Scotto34 (PCB Edition) keyboard along with default and scotto keymaps.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/scottokeebs/scotto34/info.json [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    },
+    "processor": "RP2040",
+    "url": "https://scottokeebs.com",
+    "usb": {
+        "device_version": "1.0.0",
+        "pid": "0x3334",
+        "vid": "0x534B"
+    },
+    "layouts": {
+        "LAYOUT": {
```
Comment (dunk2k) [inline keyboards/scottokeebs/scotto34/info.json:38]: Utterly optional, but changing `LAYOUT` to `LAYOUT_split_3x5_2` will allow community layouts to automatically be assigned (namely miryoku).
Would have to apply this change to all keymap.c files also.

Comment (joe-scotto) [reply keyboards/scottokeebs/scotto34/info.json]: I like that change, I added it to the board and will be for all the other ones you've been tagging :)

File: keyboards/scottokeebs/scotto34/info.json [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+            "GP10",
+            "GP11",
+            "GP12",
+            "GP13",
+            "GP18",
+            "GP19",
+            "GP20",
+            "GP21",
+            "GP22"
+        ],
```
Comment (filterpaper) [inline keyboards/scottokeebs/scotto34/info.json:27]:
```suggestion
        "cols": ["GP9", "GP10", "GP11", "GP12", "GP13", "GP18", "GP19", "GP20", "GP21", "GP22"],
```

File: keyboards/scottokeebs/scotto34/info.json [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        "cols": ["GP9", "GP10", "GP11", "GP12", "GP13", "GP18", "GP19", "GP20", "GP21", "GP22"],
+        "rows": ["GP0", "GP1", "GP14", "GP15"]
+    },
+    "processor": "RP2040",
+    "url": "https://scottokeebs.com",
+    "usb": {
+        "device_version": "1.0.0",
+        "pid": "0x3334",
+        "vid": "0x534B"
+    },
```
Comment (dunk2k) [inline keyboards/scottokeebs/scotto34/info.json:25]: Move legacy config.h and rules.mk entries to data driven model.
Could you add the following:
```
    "ws2812": {
        "pin": "GP23",
        "driver": "vendor"
    },
    "rgblight": {
        "led_count": 1
    },
```

File: keyboards/scottokeebs/scotto34/keymaps/scotto/config.h [hunk 4]
```diff
@@ -0,0 +1,27 @@
+/*
```
Comment (dunk2k) [inline keyboards/scottokeebs/scotto34/keymaps/scotto/config.h:1]: Would you be able to:
1. remove the `WS2812_DI_PIN` and `RGBLED_NUM` definitions, then
2. move this file into `default` keymap directory

File: keyboards/scottokeebs/scotto34/keymaps/scotto/rules.mk [hunk 5]
```diff
@@ -0,0 +1,3 @@
+TAP_DANCE_ENABLE = yes
```
Comment (dunk2k) [inline keyboards/scottokeebs/scotto34/keymaps/scotto/rules.mk:1]: Would you be able to:
1. remove `RGBLIGHT_ENABLE` and `WS2812_DRIVER` definitions/lines
2. move `this file into `default` keymap directory

File: keyboards/scottokeebs/scotto34/keymaps/scotto/config.h (hunk 4 above)
Comment (joe-scotto) [reply keyboards/scottokeebs/scotto34/keymaps/scotto/config.h:1]: Done in a662ed2.

File: keyboards/scottokeebs/scotto34/info.json (hunk 3 above)
Comment (joe-scotto) [reply keyboards/scottokeebs/scotto34/info.json:25]: Done in a662ed2.

File: keyboards/scottokeebs/scotto34/info.json [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+    "ws2812": {
+        "pin": "GP23",
+        "driver": "vendor"
+    },
+    "rgblight": {
+        "led_count": 1
+    },
```
Comment (dunk2k) [inline keyboards/scottokeebs/scotto34/info.json:32]: one more thing 🙊
`"community_layouts": ["split_3x5_2"],`

seeing as this keyboard is coming new to QMK 😉

File: keyboards/scottokeebs/scotto34/info.json [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+    "keyboard_name": "Scotto34 (PCB Edition)",
+    "maintainer": "joe-scotto",
+    "bootloader": "rp2040",
+    "diode_direction": "COL2ROW",
+    "features": {
+        "bootmagic": true,
+        "command": false,
+        "console": false,
+        "extrakey": true,
+        "mousekey": true,
```
Comment (dunk2k) [inline keyboards/scottokeebs/scotto34/info.json:12]: `"rgblight" : true,`

File: keyboards/scottokeebs/scotto34/info.json (hunk 6 above)
Comment (joe-scotto) [reply keyboards/scottokeebs/scotto34/info.json:32]: Added.

**Comment by dunk2k** [comment]
last one (he says)
`keymaps/scotto` is missing a `rules.mk` containing `TAP_DANCE_ENABLE = yes`

`TAP_DANCE_ENABLE` really is a keymap level config item, rather than at the keyboard level, so it'll need to be present for every keymap that has tapdance code 😉

**Comment by joe-scotto** [comment]
@dunk2k 


> `keymaps/scotto` is missing a `rules.mk` containing `TAP_DANCE_ENABLE = yes`

I'm seeing it on the PR and in my latest pull locally, are you sure it's actually missing?

**Comment by dunk2k** [comment]
Apologies, you're correct! it's the `default` keymap that's missing said `rules.mk` file.

**Review by dunk2k — APPROVED** [review]

**Review by waffle87 — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Pull request merged**